### DISCUSSION TITLE
LibreY MUST be HTTPS in order for it to work. Even your testing local site needs an SSL. Or so it seems.

### DISCUSSION THREAD

**Discussion by rjc3rd** [Ahwxorg/LibreY#163] [General]
I was testing my instance of LibreY on a VM at first. I also installed a self-signed SSL to it from the beginning. I don't even know why I did that on that instance, since it's a local VM on the same machine as I was editing. Everything was working fine, but it was a hassle to make a change then copy the files over to the VM over and over for testing.

I went ahead and installed a LEMP server on my local machine so I could just save the files in VSCodium and open the browser to test on the localhost. But I didn't bother to install the self-signed SSL on the localhost.

The site would load and I could do searches to test that functionality, but I couldn't save settings. If I clicked save it just went back to the index page with no settings saved. I even up searched for how to upgrade from php8.1 to php8.3 on Linux Mint. That didn't solve the problem.

I figured it had to be the SSL. Sure enough, it works now.

The moral of the story... You have to have a SSL in order to run the LibreY instance, even on the local test machine in order to test all the features.

https://www.howtogeek.com/devops/how-to-create-and-use-self-signed-ssl-on-nginx/
This is the link I used to install the self-signed SSL.

**Comment by davidovski** [comment]
Is this a browser specific issue when it comes to cookies? I've only used http when doing local development with firefox and haven't had such an issue.

**Reply by Ahwxorg** [reply]
Same, no issues here either. Using plain old Firefox.

**Comment by rjc3rd** [comment]
Could be. I use LibreWolf hardend with arkenfox-user.js and a nice little list of extensions. Actually a lot of sites break on my machine. But this one is all first party and no Javascript and such. I bet canvasblocker or privacy badger had something to do with it.

**Reply by Ahwxorg** [reply]
Might be your development stack too, what software do you use for that?

**Comment by rjc3rd** [comment]
@Ahwxorg Well, it's virtually the same on the VM as my local machine.

My VM is Debian12 headless running NGINX and php8.2 and a self-signed SSL.

My local machine is LinuxMint 21.3 running NGINX. At first LM repository installed php8.1. I thought there might be a difference in the PHP so I added a PPA and installed php8.3. Other than upgrading PHP, I didn't troubleshoot the issue other than to install the self-signed SSL. It instantly worked as it should. So I figured it had to be because of the SSL.

It must be something on my end since you guys don't have SSL on your local development testing setup. I'm not sure I want to "fix" it though. I have way too much time setting up my local machine. I don't want to fool with a good thing.

**Reply by Ahwxorg** [reply]
Oh, I just installed php and the required libraries and run `php -S 127.0.0.1:8000`. That's far easier, and runs on the user account :)
